apiserver/pkg/ray/raycluster.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2023 KubeAGI.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ray
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/client-go/dynamic"
+
+	"github.com/kubeagi/arcadia/apiserver/graph/generated"
+	"github.com/kubeagi/arcadia/pkg/config"
+)
+
+func ListRayClusters(ctx context.Context, c dynamic.Interface, input generated.ListCommonInput) (*generated.PaginatedResult, error) {
+	clusters, err := config.GetRayClusters(ctx, nil, c)
+	if err != nil {
+		return nil, err
+	}
+
+	var results = make([]generated.PageNode, 0, len(clusters))
+	for index, cluster := range clusters {
+		// skip if keyword not in cluster name
+		if input.Keyword != nil {
+			if !strings.Contains(cluster.Name, *input.Keyword) {
+				continue
+			}
+		}
+
+		results = append(results, &generated.RayCluster{
+			Index:         index,
+			Name:          cluster.Name,
+			HeadAddress:   &cluster.HeadAddress,
+			DashboardHost: &cluster.DashboardHost,
+			PythonVersion: &cluster.PythonVersion,
+		})
+	}
+
+	return &generated.PaginatedResult{
```

Comment:
Does `RayCluster` need paging?

Reply:
I don't think so. They only need the whole list of the ray clusters .